packages/compiler/package.json

```
@@ -7,38 +7,51 @@
   "homepage": "https://astro.build",
   "version": "1.4.2",
   "scripts": {
-    "build": "tsc -p . && esbuild node/sync.cts --bundle --format=cjs --platform=node --outfile=node/sync.cjs"
+    "build": "tsup"
   },
-  "main": "./node/index.js",
-  "types": "./node",
+  "main": "./dist/node/index.js",
+  "types": "./dist/shared/types.d.ts",
   "repository": {
     "type": "git",
     "url": "https://github.com/withastro/compiler.git"
   },
+  "files": [
+    "dist",
+    "types.d.ts",
+    "utils.d.ts",
+    "sync.d.ts"
+  ],
   "exports": {
     ".": {
-      "browser": "./browser/index.js",
-      "import": "./node/index.js",
-      "default": "./browser/index.js"
+      "types": "./dist/node/index.d.ts",
+      "browser": "./dist/browser/index.js",
+      "import": "./dist/node/index.js",
+      "require": "./dist/node/index.cjs",
+      "default": "./dist/browser/index.js"
     },
     "./sync": {
-      "import": "./node/sync.cjs",
-      "default": "./node/sync.cjs"
+      "types": "./dist/node/sync.d.ts",
+      "import": "./dist/node/sync.js",
+      "require": "./dist/node/sync.cjs",
+      "default": "./dist/node/sync.js"
     },
     "./utils": {
-      "browser": "./browser/utils.js",
-      "import": "./node/utils.js",
-      "default": "./browser/utils.js"
+      "types": "./dist/node/utils.d.ts",
+      "browser": "./dist/browser/utils.js",
+      "import": "./dist/node/utils.js",
+      "require": "./dist/node/utils.cjs",
+      "default": "./dist/browser/utils.js"
     },
-    "./astro.wasm": "./astro.wasm",
-    "./types": "./types.d.ts"
+    "./astro.wasm": "./dist/astro.wasm",
+    "./types": "./dist/shared/types.d.ts"
```

Not strictly required but I like to make sure this is always exposed. Surprised we didn't already have that!

```suggestion
    "./package.json": "./package.json"
```